RFCs/rfc-0007-zkPassport.md

```diff
@@ -0,0 +1,106 @@
+
+Using this information directly online would expose passport information online, which would not be viable.
+
+However, with ZKPs, users could provide proofs about their passport information, without exposing the passport information itself.
+
+Additionally, because passports following this standard support NFC, which is present on most smartphones, users can obtain this data themselves from their own passports relatively easily. Here for example is an open library that reads passports for iOS[[3](https://github.com/AndyQ/NFCPassportReader)].
+
+Properties stored in the standard include:
+* documentType
+* documentSubType
+* documentNumber
+* issuingAuthority
+* documentExpiryDate
+* dateOfBirth
+* gender
+* nationality
+* lastName
+* firstName
+* names
+* placeOfBirth
+* residenceAddress
+* phoneNumber
+
+## Objectives
+
+We propose a grant taking advantage of this, implementing, at a high level, the following:
+
+1. An iOS & Android App that uses NFC to allow users to get the data from their own passports, and store that data in their MINA wallet for secure use by web3 applications through the (in-progress) Mina Attestation API Standard.
+2. An o1js library that uses the Mina Attestation API Standard to create proofs of various components of one's zkPassport identity, for use in proof-of-unique-user and proof of nationality & age use cases.
+
+## Motivation and Rationale
+
+This would unlock many use cases for web3 and web2 applications connected to Mina.
+
+For applications on Mina, they can directly take advantage of this data through nullifiers for proof-of-uniqueness and other identity related proofs (see both scenarios below).
+
+For other chains, using Mina's zk state bridge, they could also leverage these proofs in their applications.
+
+Web2 applications may find use in this functionality as well. This seems particularly relevant as AI systems become increasingly powerful, and the risk this could pose to new and existing digital systems being overwhelmed by fake users.
+
+## Scenarios and Use Cases
+
+### Example Scenario 1: Proof-of-Uniqueness
+
+| Aspect           | Description |
+|------------------|-------------|
+| **Description**  | A zkApp wants a user to prove they are unique. |
+| **Requirements** | An implementation of zkPassport in the Attestation API Standard (see objectives), plus an on-chain nullifier that users can use with the zkApp. |
+| **Expected Outcome** | Via leveraging a simple library, users can make zkApps that are sybil-resistant. |
+| **Impact Analysis** | This would unlock many applications, such as one-person-one-vote, one-person-one account, Sybil-Resistant airdrops, etc. |
+
+### Example Scenario 2: Proof-of-nationality
+
+| Aspect           | Description |
+|------------------|-------------|
+| **Description**  | A zkApp wants a user to prove facts about what country they are / are not from. |
+| **Requirements** | An implementation of zkPassport in the Attestation API Standard (see objectives). |
```

Comment:
Another requirement would be to use the Attestation API to recursively prove many properties of a credential. For example `Prove(Prove(nationality===UK) & Prove(documentExpiryDate>currentBlockHeight))`.

Reply:
I think it will be good to add a link to the mentioned MINA Attestation API.